resources/components/base/Tabs.vue

```diff
@@ -26,6 +27,17 @@ module.exports = {
 		};
 	},
 
+	watch: {
+		selectedIndex: function ( newIndex ) {
+			this.setTabState( newIndex );
+			this.$emit( 'tab-change', this.tabs[ this.selectedIndex ] );
+		}
+	},
+
+	mounted: function () {
+		this.setTabState( this.selectedIndex );
+	},
+
 	methods: {
```

Comment:
I find it a bit counter-intuitive that "methods" is so low in the default order. I would have expected it to be above lifecycle hooks, not below. The order @egardner suggested initially on [the Gerrit thread about this](https://gerrit.wikimedia.org/r/c/mediawiki/extensions/NearbyPages/+/582265/9/resources/ext.nearby.scripts/Card.vue#16) also put methods before lifecycle hooks (but otherwise matches the canonical one).

What should we do here? I'm inclined to stick with the canonical order for consistency, even though it's a bit weird. It's only an issue when a component has both "methods" and either lifecycle hooks or "watch", and that's not all that common.

Reply:
Can we enforce a different order with a linter? I imagine the answer is yes.

If so, we can do what we think is best here. I agree that it's nice to define methods before lifecycle hooks, because ideally your lifecycle hooks are mostly calling existing methods anyway instead of containing a bunch of custom logic. Plus you probably want the methods closer to the template where they may be referenced.

Reply:
Yes, we can configure the linter to enforce whatever order we want. I'll amend this PR to make it methods above lifecycle hooks (and also above watch)